Code/GraphMol/ChemReactions/catch_tests.cpp

```diff
@@ -2228,3 +2228,27 @@ TEST_CASE("Structural fingerprints values") {
     CHECK(TanimotoSimilarity(*fp1, *fp2) == 0.4);
   }
 }
+
+TEST_CASE(
+    "Github #6015: react_idx property") {
+  SECTION("Ensure that atoms are marked with their reactant idx") {
+    std::unique_ptr<ChemicalReaction> rxn{
+        RxnSmartsToChemicalReaction("[C:1][O:2].[N:3][S:4]>>[C:1][O:2][N:3][S:4]")};
+    REQUIRE(rxn);
+    rxn->initReactantMatchers();
+    ROMOL_SPTR mol1(SmilesToMol("CO"));
+    ROMOL_SPTR mol2(SmilesToMol("NSP"));
+    std::vector<ROMOL_SPTR> reactants{mol1, mol2};
+    REQUIRE(reactants.size() == 2);
+    REQUIRE(reactants[0]);
+    REQUIRE(reactants[1]);
+    
+    auto products = rxn->runReactants(reactants);
+    REQUIRE(products.size() == 1);
+    CHECK(products[0][0]->getAtomWithIdx(0)->getProp<unsigned int>("react_idx") == 0);
+    CHECK(products[0][0]->getAtomWithIdx(1)->getProp<unsigned int>("react_idx") == 0);
+    CHECK(products[0][0]->getAtomWithIdx(2)->getProp<unsigned int>("react_idx") == 1);
+    CHECK(products[0][0]->getAtomWithIdx(3)->getProp<unsigned int>("react_idx") == 1);
+    CHECK(products[0][0]->getAtomWithIdx(4)->getProp<unsigned int>("react_idx") == 1);
```

Comment:
Please add a test with a reaction that adds an atom to the product (or modify this reaction to also add an atom) and test that the atom arising from that does not have the `react_idx` property set.